src/task_2.1/index.ts

```diff
@@ -0,0 +1,111 @@
+import express, { Request, Response, NextFunction } from 'express';
+import uuid from 'uuid/v1';
+import sortBy from 'lodash.sortby';
+import * as Joi from '@hapi/joi';
+import { createValidator } from 'express-joi-validation'
+import UserModel from './models/UserModel';
+import { PORT } from './constants';
+import { userCreateValidationSchema, userUpdateValidationSchema } from './validation';
+
+const users: UserModel[] = [];
+
+const validator = createValidator()
+
+const app = express();
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }))
+
+app.param('id', (req: Request, res: Response, next: NextFunction, value) => {
+    
+    next();
+});
+
+app.get('/user/:id', (req: Request, res: Response, next: NextFunction) => {
+    const { id } = req.params;
+    const user: UserModel = users.find(v => v.id === id && !v.isDeleted);
+
+    if (user) {
+        res.json(user);
+    } else {
+        res.status(400).json(`No user with id ${id} has been found`);
+    }
+
+    next();
+})
+
+app.post(
```

Comment:
if you are creating REST service, you can use more semantic methods for that. Eg:
"app.route('/products')
        .get(products.list);
        .post(products.create);

app.route('products/:id')
        .get(product.get);
        .delete(product.delete);
        .put(product.update);"

Reply:
Thank you, got it